### PR TITLE
feat: Add Jest and client-side smoke testing

### LINKED ISSUE #228: Add Jest and write a simple smoke test for R3F

### PR DESCRIPTION
**Pull request by kanyesthaker**
### Description of Changes
- Add Jest and `@react-three/test-renderer` as testing utilities
- Write basic smoke tests for the client with/without Three components
- Note that the React Three test renderer only works on children of the `<Canvas />`, so we will use traditional testing for the rest of the project

Note: I do get some warning related to using `ReactDOM.render` as part of the test due to deprecation between React 17 and 18. This should not impact functionality but can add noise. Open to suggestions. I also get a warning about using `act(...)`, even though I do not use the `await act` pattern anywhere; I believe this is because the React Fiber testing library may be outdated.

### Link to Issue
closes #228 

### Testing Procedure
- Wrote a smoke test for Scene that simply checks that the Scene component renders with nonzero elements
- Wrote a smoke test for a simple glTF that asserts that the glTF renders
image

### CODE REVIEW

File: src/scene/models/__tests__/GLTFModel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  it('should fetch an object from storage and render a group', async () => {
+    const renderer = await ReactThreeTestRenderer.create(
+      <GLTFModel
+        modelPath={TEST_CUBE_PATH}
+        modelPosition={[0, 1, 2]}
+        modelScale={1.5}
+      />,
+    );
+    const allChildren = renderer.scene.children;
+    expect(allChildren.length).toEqual(1);
```
Comment (kanyesthaker) [inline src/scene/models/__tests__/GLTFModel.test.tsx:22]: @samjgorman let me know if this is idiomatic, i.e. `toEqual` versus `toBe`

**Review by samjgorman — CHANGES_REQUESTED** [review]
Looking good! Just have some more requests to help set up jest in our application! 

1. Can you install `@types/jest `to our project's dependencies? This will help us write & type future jest tests
2. Let's create a barebones` jest.config.ts` file such that we'll have a place to modify & tweak jest settings as time goes on.  For now, we can just import the Jest defaults, like this...



````typescript

import type {Config} from 'jest';
import {defaults} from 'jest-config';

const config: Config = {
  moduleFileExtensions: [...defaults.moduleFileExtensions, 'mts'],
};

export default config;

````

3.  Mind creating a Notion doc or readme that documents the command that we can run to run these Jest tests?

File: src/scene/__tests__/Scene.test.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+
+import { MockResizeObserver } from '../../common/utils/tests';
+import Scene from '../Scene';
+
+window.ResizeObserver = MockResizeObserver;
+
+describe('Scene', () => {
+  it('should render some elements', async () => {
```
[the remark this answers is not in the document]
Comment (kanyesthaker) [reply src/scene/__tests__/Scene.test.tsx:10]: the intent behind this PR is to set up Jest and to add a smoke test to make sure that the _testing library_ itself is working. Smoke tests just do basic things like "ensure that the page does not crash when loaded" -- I'll leave the act of actually writing unit tests to a future PR.

File: src/scene/__tests__/Scene.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+
+import { MockResizeObserver } from '../../common/utils/tests';
+import Scene from '../Scene';
+
+window.ResizeObserver = MockResizeObserver;
```
[the remark this answers is not in the document]
Comment (kanyesthaker) [reply src/scene/__tests__/Scene.test.tsx:7]: Without mocking ResizeObserver rendering `<Scene />` will fail with 
```
This browser does not support ResizeObserver out of the box. See: https://github.com/react-spring/react-use-measure/#resize-observer-polyfills
```
I believe the React Fiber `<Canvas />` component uses ResizeObserver by default. I didn't bother digging too deep.

[the remark this answers is not in the document]
Comment (kanyesthaker) [reply src/scene/__tests__/Scene.test.tsx:7]: Actually yes -- the Fiber `<Canvas />` component uses `react-use-measure` which requires `ResizeObserver`, which is not available in our testing environment https://docs.pmnd.rs/react-three-fiber/api/canvas

File: src/scene/__tests__/Scene.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (samjgorman) [reply src/scene/__tests__/Scene.test.tsx:10]: That's alright with me

**Review by samjgorman — APPROVED** [review]
Great job getting this test infra in!

**Pull request merged**